src/Symfony/Component/Form/ChoiceList/ArrayChoiceList.php

```diff
@@ -236,7 +236,11 @@ private function castableToString(array $choices, array &$cache = array())
                 continue;
             } elseif (!is_scalar($choice)) {
                 return false;
-            } elseif (isset($cache[$choice])) {
+            }
+
+            $choice = false === $choice ? '0' : (string) $choice;
```

Comment:
I like your idea (perhaps that allows render `"false"` instead of `"0"` for boolean values) but we need to detect duplicates now when (`0` and `false`) or (`1` and `true`) coexist in the same choices, else it breaks the current behavior when [above](https://github.com/yceruto/symfony/blob/65d1af6a75db795fd04e1a322c7358c29b18b50a/src/Symfony/Component/Form/ChoiceList/ArrayChoiceList.php#L211) these values are casted to (string) again. Maybe in other PR ? ;)